### DISCUSSION TITLE
Cross Cluster Replication backward compatibility during cluster upgrades

### DISCUSSION THREAD

**Discussion by isuru-instaclustr** [opensearch-project/cross-cluster-replication#1392] [Q&A] [answered]
Is there a backward compatibility matrix documented for opensearch cross cluster replication plugin similar to [elasticsearch](https://www.elastic.co/guide/en/elasticsearch/reference/current/xpack-ccr.html) ? Can we safely assume that the replication will work with the previous major/minor versions always during a cluster upgrade if we follow the sequence of upgrading the follower cluster first? I have cross posted this in [OpenSearch forum](https://forum.opensearch.org/t/cross-cluster-replication-backward-compatibility-during-cluster-upgrades/19692) as well in case someone might know the answer. thanks for the help!

**Comment by DarshitChanpura** [comment]
Tagging the maintainers to help answer this question. @saikaranam-amazon @gbbafna @soosinha @ankitkala @krishna-ggk

**Comment by pajuric** [comment]
Please keep in the loop on this, in addition to Instaclustr.

**Comment by ankitkala** [comment, marked as answer]
Hi, there isn't any compatibility matrix for cross cluster replication. Similar to rolling upgrade, last minor version should be compatible with next major version. 

As long as you're upgrading follower cluster first, it should not break.

**Reply by isuru-instaclustr** [reply]
thanks!

**Reply by isuru-instaclustr** [reply]
@ankitkala  should we follow the same pattern of upgrading followers first during minor version upgrades within the same major version? or is it safe to assume that there's both forward and backward compatibility between followers and leaders in the same major version but different minor versions.

**Reply by ankitkala** [reply]
We recommend always upgrading follower first.

**Comment by soosinha** [comment]
You can file a documentation issue to get this documented: https://github.com/opensearch-project/documentation-website/issues/

**Reply by isuru-instaclustr** [reply]
thanks! I have raised [this](https://github.com/opensearch-project/documentation-website/issues/7832).
